### PR TITLE
Fix verify boilerplate

### LINKED ISSUE #23: verify-boilerplate does not tolerate copyright lines without year
/kind bug
/assign

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

copyright lines in generated files (created by the k8s code-gen) tools don't have years.
this makes the verify boilerplate check fail.

support copyright lines without year for autogenerated files.

https://github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm/pull/22#issuecomment-510644042

### PR DESCRIPTION
**Pull request by ashish-amarnath**
**What this PR does / why we need it**:
Fixes boilerplate verification in the repo.
New boilerplate verification code is from:
* https://github.com/kubernetes-sigs/cluster-api-provider-aws/tree/master/hack/boilerplate
* https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/hack/verify_boilerplate.py

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #23

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
NONE
```

/assign @neolit123

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,3 +1,17 @@
+# Copyright 2019 The Kubernetes Authors.
```
Comment (detiber) [inline Dockerfile]: It seems weird to have boilerplate on the dockerfile

File: Makefile [hunk 2]
```diff
@@ -1,3 +1,16 @@
+# Copyright 2019 The Kubernetes Authors.
```
Comment (detiber) [inline Makefile]: Same here with the Makefile

File: hack/verify_boilerplate.py [hunk 3]
```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env python
```
Comment (chuckha) [inline hack/verify_boilerplate.py]: Why is this replacing the go one?

**Comment by vincepri** [comment]
Can we use the same structure that's in Cluster API? 

- `hack/verify-boilerplate.sh` https://github.com/kubernetes-sigs/cluster-api/blob/master/hack/verify-boilerplate.sh
- `hack/boilerplate/*` https://github.com/kubernetes-sigs/cluster-api/tree/master/hack/boilerplate

File: Dockerfile (hunk 1 above)
Comment (ashish-amarnath) [reply Dockerfile]: There are boilerplate templates for `Dockerfile` and the python script, copied from capa (which was copied from another kubernetes repo), enforces this.  So 🤷‍♂

File: Makefile (hunk 2 above)
Comment (ashish-amarnath) [reply Makefile]: There are boilerplate templates for `Makefile` and the python script, copied from capa (which was copied from another kubernetes repo), enforces this. Also, capa `Makefile` has the boilerplate stuff.

File: hack/verify_boilerplate.py (hunk 3 above)
Comment (ashish-amarnath) [reply hack/verify_boilerplate.py]: Discussed in #23. The go one has a bug. Instead of fixing the bug, we are trying to use what capa is using.

**Comment by ashish-amarnath** [comment]
@vincepri that python script also seems to have a bug. It failing to identify `/Users/ashisha/Work/kubernetes/src/cluster-api-bootstrap-provider-kubeadm/api/v1alpha1/zz_generated.deepcopy.go` as a generated file.
```
./hack/boilerplate/boilerplate.py /Users/ashisha/Work/kubernetes/src/cluster-api-bootstrap-provider-kubeadm/api/v1alpha1/zz_generated.deepcopy.go -v
Header in /Users/ashisha/Work/kubernetes/src/cluster-api-bootstrap-provider-kubeadm/api/v1alpha1/zz_generated.deepcopy.go does not match reference, diff:

--- reference
+++ /Users/ashisha/Work/kubernetes/src/cluster-api-bootstrap-provider-kubeadm/api/v1alpha1/zz_generated.deepcopy.go
@@ -1,5 +1,5 @@
 /*
-Copyright YEAR The Kubernetes Authors.
+Copyright The Kubernetes Authors.

 Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.

/Users/ashisha/Work/kubernetes/src/cluster-api-bootstrap-provider-kubeadm/api/v1alpha1/zz_generated.deepcopy.go
```

**Comment by ashish-amarnath** [comment]
We seem to have a fix of ignoring the generated file in master. Closing the PR.

**Pull request closed without merging**